lightning/src/routing/router.rs

```diff
@@ -1498,6 +1501,55 @@ where L::Target: Logger {
 		}
 	}
 
+	// Step (10).
+	// Add a random 'shadow route' offset to the CLTV expiry delta values to improve privacy.
+	//
+	// Start from the payee
+	let mut cur_node_id = payee_node_id;
+	let mut shadow_ctlv_expiry_delta_offset: u32 = 0;
+
+	let mut prng = ChaCha20::new(random_bytes, &[0; 8]);
+	let mut random_index_bytes = [0u8; 8];
+	let mut random_index: usize;
+
+	// Do a three-hop random walk
+	for _random_hop in 0..3 { 
+		if let Some(cur_node) = network_nodes.get(&cur_node_id) {
+			// Randomly choose the next hop
+			prng.process_in_place(&mut random_index_bytes);
+			random_index = usize::from_be_bytes(random_index_bytes).wrapping_rem(cur_node.channels.len());
```

Comment:
Ah, good point. I think this would currently never be the case since no channels would mean that no path would be found and the function would return in https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/routing/router.rs#L1396. However, its definitely safer to check!

Reply:
Ah, good point! If I'm not mistaken `channels.len() == 0` could only ever happen when the last hop is reached through a private channel. I'll hence address this together with your next comment.

Reply:
Addressed with https://github.com/lightningdevkit/rust-lightning/commit/ea0ad9d079ba51e5cf6f780999661a1b7673351b.